### PR TITLE
[ADM-440][Report] [backend] Load Github data for leadtime for changes in Github Service

### PR DESCRIPTION
**Pull request by wangwang-zhang**
## Summary

implemented three github feignClient api interface in GithubFeignClient
- getCommitInfo
- getPullRequestCommitInfo
- getPullRequestListInfo
besides , also optimised backend client dto directory structure

### CODE REVIEW

File: backend/src/main/java/heartbeat/client/GithubFeignClient.java
Comment (wangwang-zhang) [inline backend/src/main/java/heartbeat/client/GithubFeignClient.java:1]: these are three newly implemented feignClient api interface.

File: backend/src/main/java/heartbeat/service/source/github/GithubService.java [hunk 1]
```diff
@@ -2,9 +2,9 @@
 
 import feign.FeignException;
 import heartbeat.client.GithubFeignClient;
-import heartbeat.client.dto.GithubOrganizationsInfo;
-import heartbeat.client.dto.GithubRepos;
-import heartbeat.controller.source.vo.GithubResponse;
+import heartbeat.client.dto.codebase.github.GithubOrganizationsInfo;
```
Comment (guzhongren) [inline backend/src/main/java/heartbeat/service/source/github/GithubService.java:5]: PLZ rename all `Github` to `GitHub`

File: backend/src/main/java/heartbeat/util/StringToEnumConverter.java [hunk 2]
```diff
@@ -1,6 +1,6 @@
 package heartbeat.util;
 
-import heartbeat.controller.board.vo.request.BoardType;
+import heartbeat.controller.board.dto.request.BoardType;
```
Comment (guzhongren) [inline backend/src/main/java/heartbeat/util/StringToEnumConverter.java:3]: this file `covert` method has been implamented by `heartbeat.controller.board.dto.request.BoardType`, 
PLZ refactor to remove `covert`

File: backend/src/main/java/heartbeat/controller/source/dto/GithubResponse.java [hunk 3]
```diff
@@ -1,4 +1,4 @@
-package heartbeat.controller.source.vo;
+package heartbeat.controller.source.dto;
```
Comment (guzhongren) [inline backend/src/main/java/heartbeat/controller/source/dto/GithubResponse.java:1]: PLZ help to rename this file name to `GitHubResponse`, and next file `GitHubController`

File: backend/src/main/java/heartbeat/client/GitHubFeignClient.java [hunk 4]
```diff
@@ -14,19 +14,19 @@
 import java.util.List;
 
 @FeignClient(name = "githubFeignClient", url = "${github.url}")
-public interface GithubFeignClient {
+public interface GitHubFeignClient {
 
 	@GetMapping(path = "/user/orgs")
 	@ResponseStatus(HttpStatus.OK)
-	List<GithubOrganizationsInfo> getGithubOrganizationsInfo(@RequestHeader("Authorization") String token);
+	List<GitHubOrganizationsInfo> getGithubOrganizationsInfo(@RequestHeader("Authorization") String token);
```
Comment (guzhongren) [inline backend/src/main/java/heartbeat/client/GitHubFeignClient.java:21]: Rename getGithubOrganizationsInfo

File: backend/src/main/java/heartbeat/client/GitHubFeignClient.java [hunk 5]
```diff
@@ -14,19 +14,19 @@
 import java.util.List;
 
 @FeignClient(name = "githubFeignClient", url = "${github.url}")
-public interface GithubFeignClient {
+public interface GitHubFeignClient {
```
Comment (guzhongren) [inline backend/src/main/java/heartbeat/client/GitHubFeignClient.java:17]: Do I have design decoder for this client, you can seek help from guoqing

Comment (guzhongren) [reply backend/src/main/java/heartbeat/client/GitHubFeignClient.java:17]: Maybe do it in another card

**Review by guzhongren — APPROVED** [review]
Give some update on the comments

File: backend/src/main/java/heartbeat/client/GitHubFeignClient.java (hunk 5 above)
Comment (wangwang-zhang) [reply backend/src/main/java/heartbeat/client/GitHubFeignClient.java:17]: ok

**Review by guzhongren — APPROVED** [review]

**Pull request closed without merging**